core/include/cubos/core/log.hpp

```diff
@@ -5,6 +5,7 @@
 #pragma once
 
 #include <cstdlib>
+#include <memory>
```

Comment:
Avoid including this in the header file - these STL headers are really heavy on the compile times, this logging header particularly is included pretty much everywhere should it be as lean as possible.